src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -598,8 +598,7 @@ private TypeSymbol BindVariableType(CSharpSyntaxNode declarationNode, Diagnostic
             // or it might not; if it is not then we do not want to report an error. If it is, then
             // we want to treat the declaration as an explicitly typed declaration.
 
-            RefKind refKind;
-            TypeSymbol declType = BindType(typeSyntax.SkipRef(out refKind), diagnostics, out isVar, out alias);
```

Comment:
None of the other paths are skipping the ref, but I don't think they can have a ref.
The other paths are: binding deconstruction variables, declaration expressions, out vars, out variable declarations. 
The only one that can have ref type is the binding of declaration statement, from what I understand.